### PR TITLE
feat: Add ability to restore etcd backup from s3 to a cluster with a different name 

### PR DESCRIPTION
**Pull request by Aveek-Saha**
# Description
Feature: Currently the etcd backup restore from s3 assumes that the restored cluster and the original cluster share the same name.  This PR allows restoring to a cluster with a different name from the original. This change enables the deletion of the old nodes from the new cluster.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Small minor change not affecting the Ansible Role code (Github Actions Workflow, Documentation etc.)

## How Has This Been Tested?

I've tested this by restoring a backup from s3 using a cluster with a different name from the original cluster

### CODE REVIEW

File: tasks/first_server.yml [hunk 1]
```diff
@@ -127,3 +127,22 @@
...
+
+- name: Restore etcd - remove old nodes
+  ansible.builtin.shell: |
+    {{ rke2_data_path }}/bin/kubectl --kubeconfig /etc/rancher/rke2/rke2.yaml \
+    delete node {{ item }} 2>&1 || true
+  args:
+    executable: /bin/bash
+  with_items: "{{ node_names.stdout_lines | difference(groups[rke2_cluster_group_name]) }}"
+  changed_when: false
+  when: inventory_hostname != item and (do_etcd_restore is defined or do_etcd_restore_from_s3 is defined)
```
Comment (MonolithProjects) [inline tasks/first_server.yml:148]: Please correct the yaml syntax and add an empty line to the file.

**Review by MonolithProjects — CHANGES_REQUESTED** [review]

File: tasks/first_server.yml (hunk 1 above)
Comment (Aveek-Saha) [reply tasks/first_server.yml]: Added an new line to the end of the file

Comment (MonolithProjects) [reply tasks/first_server.yml]: thx

**Review by MonolithProjects — APPROVED** [review]
LGTM

**Pull request merged**